[CI] CoreWithMultipleProjectsClientYamlTestSuiteIT test {yaml=search.vectors/41_knn_search_bbq_hnsw/Test index configured rescore vector score consistency} failing

**Build Scans:**
- [elasticsearch-periodic #7328 / openjdk21_entitlements_false_checkpart2_java-matrix](https://gradle-enterprise.elastic.co/s/pntjqfazhc4ak)
- [elasticsearch-pull-request #65412 / part-2](https://gradle-enterprise.elastic.co/s/v5l6dhlrtq42e)
- [elasticsearch-pull-request #65329 / part-2](https://gradle-enterprise.elastic.co/s/tsqbvdntcgzlo)

**Reproduction Line:**
```
./gradlew ":x-pack:qa:multi-project:core-rest-tests-with-multiple-projects:yamlRestTest" --tests "org.elasticsearch.multiproject.test.CoreWithMultipleProjectsClientYamlTestSuiteIT" -Dtests.method="test {yaml=search.vectors/41_knn_search_bbq_hnsw/Test index configured rescore vector score consistency}" -Dtests.seed=E639BE0AF4AC13A2 -Dtests.jvm.argline="-Des.entitlements.enabled=false" -Dtests.locale=en-CH -Dtests.timezone=Europe/Saratov -Druntime.java=21
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.multiproject.test.CoreWithMultipleProjectsClientYamlTestSuiteIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(test%20%7Byaml%3Dsearch.vectors%2F41_knn_search_bbq_hnsw%2FTest%20index%20configured%20rescore%20vector%20score%20consistency%7D),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: Failure at [search.vectors/41_knn_search_bbq_hnsw:594]: 
Expected: <6.3367105>
     but: was <6.392275>
```

**Issue Reasons:**
- [main] 3 failures in test test {yaml=search.vectors/41_knn_search_bbq_hnsw/Test index configured rescore vector score consistency} (1.6% fail rate in 192 executions)
- [main] 2 failures in step part-2 (3.8% fail rate in 53 executions)
- [main] 2 failures in pipeline elasticsearch-pull-request (3.8% fail rate in 52 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -368,9 +368,6 @@ tests:
 - class: org.elasticsearch.xpack.downsample.DataStreamLifecycleDownsampleDisruptionIT
   method: testDataStreamLifecycleDownsampleRollingRestart
   issue: https://github.com/elastic/elasticsearch/issues/123769
-- class: org.elasticsearch.multiproject.test.CoreWithMultipleProjectsClientYamlTestSuiteIT
-  method: test {yaml=search.vectors/41_knn_search_bbq_hnsw/Test index configured rescore vector score consistency}
-  issue: https://github.com/elastic/elasticsearch/issues/125902
 - class: org.elasticsearch.xpack.test.rest.XPackRestIT
   method: test {p0=ml/start_data_frame_analytics/Test start given dest index is not empty}
   issue: https://github.com/elastic/elasticsearch/issues/125909

